lib/launcher.js

```diff
@@ -255,11 +255,12 @@ class Launcher {
         let config = this.configParser.getConfig()
         let debug = caps.debug || config.debug
         let rid = this.getRunnerId(cid)
+        let processNumber = this.getNumberOfRunningInstances() + 1
```

Comment:
You're right.  I thought I could just use the number of running instances instead of adding state, but I didn't think about what would happen after a process terminates and the count is decremented.  So if maxInstances is 10 the 11th process would have id 10 and 10 may still be running.